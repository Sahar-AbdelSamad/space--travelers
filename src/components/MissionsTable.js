import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import MissionList from './MissionList';
import { fetchDataMissions } from '../redux/missions/missions';

const MissionsTable = () => {
  const missions = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDataMissions());
  }, []);

  return (
    <div>
      <table className="table table-striped table-hover">
        <tr className="fs-5">
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
        <tbody>
          {
            missions.map((mission) => (
              <MissionList
                key={mission.mission_id}
                id={mission.mission_id}
                missionName={mission.mission_name}
                description={mission.description}
              />
            ))
          }
        </tbody>
      </table>
    </div>
  );
};
export default MissionsTable;
