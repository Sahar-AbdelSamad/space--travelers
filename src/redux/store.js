import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer, { fetchRockets } from './rockets/rocketsSlice';
import missionsReducer, { fetchMissions } from './missions/missionsSlice';
import mobileMenuReducer from './mobileMenu/mobileMenuSlice';

export const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
    mobileMenu: mobileMenuReducer,
  },
});

store.dispatch(fetchRockets());
store.dispatch(fetchMissions());

export default store;
