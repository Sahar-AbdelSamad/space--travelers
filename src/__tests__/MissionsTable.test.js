import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MissionsTable from '../components/MissionsTable';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionsTable />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
