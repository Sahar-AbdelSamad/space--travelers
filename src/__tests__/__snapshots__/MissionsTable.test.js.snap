// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    className="table-bordered table-striped table-hover"
  >
    <thead>
      <tr
        className="fs-5"
      >
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
